### PR TITLE
Fix various issues in perlfaq4

### PR DESCRIPTION
**Pull request by briandfoy**
This is based on the list of questions in #101.

### CODE REVIEW

**Review by davorg — DISMISSED** [review]

File: lib/perlfaq4.pod [hunk 1]
```diff
@@ -273,32 +273,63 @@ are left as an exercise to the inclined reader.
...
-the programmer says:
-
-    if ("\020\020" & "\101\101") {
-        # ...
-    }
+Perl's C<&> bitwise operator works on both numbers and strings,
+sometimes producing surprising results when you expected a number
+but received a string. You probably expected perl to automatically
+convert the operands to numbers like the mathematical operators would.
+Instead, perl treats string operands as a bitvectors.
```
Comment (bigpresh) [inline lib/perlfaq4.pod:280]:
```suggestion
Instead, perl treats string operands as bitvectors.
```

File: lib/perlfaq4.pod [hunk 2]
```diff
@@ -273,32 +273,63 @@ are left as an exercise to the inclined reader.
...
+
+	my($i, $j) = ( 11,   3 );   # $i & $j  # 11  &  3 -> 3
+ 	my($i, $j) = ("11",  3 );   # $i & $j  # 11  &  3 -> 3
+ 	my($i, $j) = ("11", "3");   # $i & $j  # "11"  &  "3" -> 1
+
+Remember that a perl scalar can have both string and numeric values at
+the same time. A value that starts as a string and has never encountered
+a numeric operation has no numeric value yet. Perl does this to save
+time and work so it doesn't have to decide a numeric value for a scalar
+it might never use as a number. In that case, string sematics wins. But,
```
Comment (bigpresh) [inline lib/perlfaq4.pod:308]:
```suggestion
it might never use as a number. In that case, string semantics wins. But,
```

**Review by bigpresh — CHANGES_REQUESTED** [review]
Minor typo tweaks, but looks good otherwise!

File: lib/perlfaq4.pod (hunk 1 above)
Comment (briandfoy) [reply lib/perlfaq4.pod:280]: fixed in ddd936a77b5b6f4aa41024f68d7b2e1013ecfaec (which isn't showing up yet)

File: lib/perlfaq4.pod (hunk 2 above)
Comment (briandfoy) [reply lib/perlfaq4.pod:308]: fixed in ddd936a77b5b6f4aa41024f68d7b2e1013ecfaec (which isn't showing up yet)

**Review by bigpresh — APPROVED** [review]

**Review by davorg — APPROVED** [review]

**Comment by davorg** [comment]
Who has permissions to unblock this PR (and the others that @briandfoy currently has stacked up)? I'd be happy to take on more responsibility here if that would help.

**Comment by karenetheridge** [comment]
This was merged; thanks!

**Pull request closed without merging**